OpenSim/Simulation/Model/ExpressionBasedPointToPointForce.cpp

```diff
@@ -117,10 +117,17 @@ void ExpressionBasedPointToPointForce::extendConnectToModel(Model& model)
     const string& body1Name = getBody1Name();
     const string& body2Name = getBody2Name();
 
-    _body1 =
-        static_cast<const PhysicalFrame*>(&getModel().getComponent(body1Name));
-    _body2 =
-        static_cast<const PhysicalFrame*>(&getModel().getComponent(body2Name));
+    if(getModel().hasComponent(body1Name))
+        _body1 = &(getModel().getComponent<PhysicalFrame>(body1Name));
+    else
+        _body1 = &(getModel().getComponent<PhysicalFrame>(
+            "./bodyset/" + body1Name));
+
```

Comment:
At connect stage, are we guaranteed that either of the two pathways will succeed and we're not left with nullptr?

Reply:
No `getComponent` throws if no component.

Reply:
`getComponent()` throws an exception if it can't find the component.